### ISSUE TITLE
Consider alternative compression library 

### ISSUE THREAD

**Issue by alexcroox** [rrweb-io/rrweb#381]
Initially when Pako was introduced to this library there were concerns over its huge size. Thought I’d throw a tiny, more performant alternative into the mix for future consideration:

https://github.com/101arrowz/fflate

> fflate (short for fast flate) is the fastest, smallest, and most versatile pure JavaScript compression and decompression library in existence, handily beating pako, tiny-inflate, and UZIP.js in performance benchmarks while being multiple times more lightweight.

**Comment by Yuyz0112** [comment]
Upvote this suggestion!

The only thing I'm thinking about is how to handle compatibility problems since there are already some data compressed by pako.

**Comment by alexcroox** [comment]
I’ve not looked at the data before, do you store a version number/date or similar against the compressed data?

**Comment by Yuyz0112** [comment]
@alexcroox Yes we have a version number in the data, but what's the expected behavior when we provide a new packer via `fflate`, and meet data compressed by `pako`? Should we throw an error to let users use the v1 packer?

**Comment by 101arrowz** [comment]
Hi, library author here! As mentioned in the summary, data compressed by `fflate` can be decompressed by `pako` and vice versa. This wouldn't be a breaking change for that reason.

**Comment by alexcroox** [comment]
Thanks for taking the time to let us know @101arrowz!!

**Comment by 101arrowz** [comment]
I've looked into the codebase and I think this will work and cut down bundle size while improving performance by a bit.
```patch
 package.json                     |  2 +-
 src/declarations/pako/index.d.ts | 12 ------------
 src/packer/pack.ts               |  4 ++--
 src/packer/unpack.ts             |  4 ++--
 4 files changed, 5 insertions(+), 17 deletions(-)
 delete mode 100644 src/declarations/pako/index.d.ts

diff --git a/package.json b/package.json
index 5e718a2..3c6a682 100644
--- a/package.json
+++ b/package.json
@@ -60,8 +60,8 @@
   "dependencies": {
     "@types/css-font-loading-module": "0.0.4",
     "@xstate/fsm": "^1.4.0",
+    "fflate": "^0.4.2",
     "mitt": "^1.1.3",
-    "pako": "^1.0.11",
     "rrweb-snapshot": "^1.0.1"
   }
 }
diff --git a/src/declarations/pako/index.d.ts b/src/declarations/pako/index.d.ts
deleted file mode 100644
index e92682a..0000000
--- a/src/declarations/pako/index.d.ts
@@ -1,12 +0,0 @@
-declare module 'pako/dist/pako_deflate' {
-  export const deflate: any;
-}
-
-declare module 'pako/dist/pako_inflate' {
-  export const inflate: any;
-}
-
-declare module 'pako' {
-  export const deflate: any;
-  export const inflate: any;
-}
diff --git a/src/packer/pack.ts b/src/packer/pack.ts
index f62bb92..c9c3851 100644
--- a/src/packer/pack.ts
+++ b/src/packer/pack.ts
@@ -1,4 +1,4 @@
-import { deflate } from 'pako/dist/pako_deflate';
+import { zlibSync, strFromU8, strToU8 } from 'fflate';
 import { PackFn, MARK, eventWithTimeAndPacker } from './base';
 
 export const pack: PackFn = (event) => {
@@ -6,5 +6,5 @@ export const pack: PackFn = (event) => {
     ...event,
     v: MARK,
   };
-  return deflate(JSON.stringify(_e), { to: 'string' });
+  return strFromU8(zlibSync(strToU8(JSON.stringify(_e))), true);
 };
diff --git a/src/packer/unpack.ts b/src/packer/unpack.ts
index 4d46f12..29b6826 100644
--- a/src/packer/unpack.ts
+++ b/src/packer/unpack.ts
@@ -1,4 +1,4 @@
-import { inflate } from 'pako/dist/pako_inflate';
+import { unzlibSync, strFromU8, strToU8 } from 'fflate';
 import { UnpackFn, eventWithTimeAndPacker, MARK } from './base';
 import { eventWithTime } from '../types';
 
@@ -16,7 +16,7 @@ export const unpack: UnpackFn = (raw: string) => {
   }
   try {
     const e: eventWithTimeAndPacker = JSON.parse(
-      inflate(raw, { to: 'string' }),
+      strFromU8(unzlibSync(strToU8(raw, true))),
     );
     if (e.v === MARK) {
       return e;
```

GitHub formatting is weird, but hopefully you see the changes that I made. I made the API fully compatible, but in a major version bump, you'll want to replace `zlibSync` with `deflateSync` and `unzlibSync` with `inflateSync` because those are faster and compress better but incompatible.

**Comment by 101arrowz** [comment]
Should this be closed since #448 is merged?
